### PR TITLE
Provide more info if no update is available

### LINKED ISSUE #13: Provide more info when checking for update
### Description of the Issue
When a new version of Notepad++ is released on a site, it is not available when checking for updates from installed Notepad++.

This got asked a lot on the community forum. See recent posts:
https://community.notepad-plus-plus.org/topic/20590/updating-through-npp-occurs-very-late
https://community.notepad-plus-plus.org/topic/20592/no-update-available-but-update-is-available
And some older posts:
https://community.notepad-plus-plus.org/topic/15433/update-notepad-does-not-work
https://community.notepad-plus-plus.org/topic/15430/please-update-getdownloadurl-php
https://community.notepad-plus-plus.org/topic/16919/update-stopped-working-under-7-5-9

You may read through them, sometimes there might be interesting suggestions.

I know that the auto-update is usually triggered 1-2 weeks after the initial release, in case there are major regressions that weren’t found during the internal testing. But some of the users seem confused, so it could be better to improve the information about updates.

Also, see discussion in another issue https://github.com/notepad-plus-plus/notepad-plus-plus/issues/9393

### Steps to Reproduce the Issue

1. Prerequisite: New version of Notepad++ is released on the site (7.9.2)
2. Launch the previous version of Notepad++ (7.9.1) 
3. Open from menu _? > Update Notepad++_

### Expected Behavior
A message box with text is shown:
```
No update is available.

The auto-update may not have been activated yet.
Go to the Notepad++ official website to check for a new version?
[Cancel]   [OK]
```

Click [OK] will open the browser and lead the user to the Notepad++ Download page.

Other options:

1. A message box appears saying something like "A new version is pending approval. You can download it from the official site."
2. A message box appears saying something like "A new version is unstable. Update anyway? Yes/No"
3. Someone from the community suggested splitting the releases to be alpha, beta, and RC (and maybe stable). But this seems complicated.
4. Any other suggestions?

### Actual Behavior

A message box appears saying "No update is available":
image

### Debug Information

Notepad++ v7.9.1   (64-bit)
Build time : Nov  2 2020 - 01:07:46
Path : C:\Program Files\Notepad++\notepad++.exe
Admin mode : ON
Local Conf mode : OFF
OS Name : Windows 7 Professional (64-bit) 
OS Build : 7601.24557
Current ANSI codepage : 1252
Plugins : HexEditor.dll mimeTools.dll NppConverter.dll NppExport.dll

### PR DESCRIPTION
**Pull request by mere-human**
Also, allow users to go to the website and check for updates manually.

Fix #13

### CODE REVIEW

File: src/ConfigFiles/nativeLang.xml [hunk 1]
```diff
@@ -22,7 +22,7 @@
 
 <GUP_NativeLangue name = "English">
 	<PopupMessages>
-		<MSGID_NOUPDATE>No update is available.</MSGID_NOUPDATE>
+		<MSGID_NOUPDATE>No update is available.\r\rThe auto-update may not have been activated yet.\rGo to the official website to check for a new version?</MSGID_NOUPDATE>
```
Comment (mere-human) [inline src/ConfigFiles/nativeLang.xml:25]: As far as I understood, this is not used in Notepad++.
If I try to load this in debug, then the message is displayed incorrectly:
image

It seems LF is not loaded properly from XML.

File: src/winmain.cpp [hunk 2]
```diff
@@ -1085,7 +1088,12 @@ int WINAPI WinMain(HINSTANCE hInstance, HINSTANCE, LPSTR lpszCmdLine, int)
 				string noUpdate = nativeLang.getMessageString("MSGID_NOUPDATE");
 				if (noUpdate == "")
 					noUpdate = MSGID_NOUPDATE;
-				::MessageBoxA(NULL, noUpdate.c_str(), gupParams.getMessageBoxTitle().c_str(), MB_OK);
+				int res = ::MessageBoxA(NULL, noUpdate.c_str(), gupParams.getMessageBoxTitle().c_str(), MB_OKCANCEL);
+				if (res == IDOK)
+				{
+					// This URL is better to be in config
+					::ShellExecute(NULL, TEXT("open"), TEXT("https://notepad-plus-plus.org/downloads/"), NULL, NULL, SW_SHOWNORMAL);
```
Comment (mere-human) [inline src/winmain.cpp:1095]: Didn't find a good place so it is here for now.
It could be moved to XML config or passed as a command-line argument.

**Comment by mere-human** [comment]
Now the "No update" dialog looks like this:
image

**Comment by mere-human** [comment]
This PR should be deleted since it contains the wrong email.

**Comment by mere-human** [comment]
@donho 

I contacted GitHub Support in order to delete this PR.
Could you, please, give the authorization to delete this PR here in the comments?

Or could you delete it manually from this form: https://support.github.com/contact?tags=rr-remove-data

> Hi there,
>
> Thank you for writing in to GitHub Support.
> 
> As the PR has had some conversation on it, we would need admin authorization to delete it.
> 
> Could you please ask a repository admin to authorize this request by replying to this message with an email address associated with their GitHub account? Or they could leave a comment giving authorization on the PR itself.

**Comment by donho** [comment]
> This PR should be deleted since it contains the wrong email.

I didn't see any email in this PR.

> I contacted GitHub Support in order to delete this PR.
> Could you, please, give the authorization to delete this PR here in the comments?

I can close a PR, but I cannot delete it.

> Or could you delete it manually from this form: https://support.github.com/contact?tags=rr-remove-data

I will try it now.

**Comment by mere-human** [comment]
> I didn't see any email in this PR.

I mean in commit info.
The email was set incorrectly in git config and I didn't want to expose that email.
Sorry for the inconvenience.

**Comment by donho** [comment]
Just sent a request to GitHub, hope they can remove it:

image

**Pull request closed without merging**